ateliermecanique-ws/src/main/java/com/champlain/ateliermecaniquews/customeraccountsmanagementsubdomain/businesslayer/CustomerAccountServiceImpl.java

```diff
@@ -59,5 +60,18 @@ public CustomerAccountResponseModel updateCustomerById(String customerId, Custom
         }
     }
 
+    @Override
+    public void deleteCustomerById(String customerId) {
+        CustomerAccount account = customerAccountRepository.findCustomerAccountByCustomerAccountIdentifier_CustomerId(customerId);
+
+        if (account != null){
+            customerAccountRepository.delete(account);
+            vehicleService.deleteAllVehiclesByCustomerId(customerId);
+        }
+        else{
+            //throw exception
+        }

```

Comment:
Good stuff Cristian, only thing to change would be throwing the exception here. We'll add that soon